pennylane/operation.py

```diff
@@ -1520,7 +1520,7 @@ def _obs_data(self):
         >>> print(tensor._obs_data())
         {("PauliZ", <Wires = [1]>, ()), ("PauliX", <Wires = [0]>, ())}
         """
-        obs = Tensor(self).non_identity_obs
+        obs = Tensor(self, do_queue=False).non_identity_obs
```

Comment:
Nathan will definitely cry - he wants the `do_queue` functionality to be removed since years. If we make this a short-term solution we'd have to have extremely good arguments (because of the risk that it stays).

Reply:
I can instead do this inside of a stop recording context.  

But most of the workarounds in this PR make me uncomfortable and I wish there was a better way, but we can't keep relying on the `Operator`/ `Operation` distinction to separate out things in the tape.

Reply:
I feel like we will need a "tape refactor" soon :/